src/Common/src/System/Net/NetworkInformation/UnixCommandLinePing.cs

```diff
@@ -37,6 +41,27 @@ private static string GetPingUtilityPath(bool ipv4)
             return null;
         }
 
+        // Check if found ping is symlink to busybox like alpine /bin/ping -> /bin/busybox
+        private static unsafe bool IsBusyboxPing(string pingBinary)
+        {
+            byte* buffer = stackalloc byte[4000];
+            int resultLength = ReadLink(pingBinary, buffer, 4000);
+
+            // If pingBinary is not link resultLength will be -1
+            if (resultLength > 0 && resultLength < 4000)
+            {
+                string symlinkPath = Marshal.PtrToStringAnsi((IntPtr)buffer, resultLength);
+                if (symlinkPath.EndsWith("busybox"))
```

Comment:
EndsWith does a culture-based comparison.  We probably want to pass StringComparison.Ordinal, right?